### PR TITLE
[Ahuna] Assistant Details modal

### PR DESCRIPTION
**Pull request by philipperolet**
Related [task](https://github.com/dust-tt/dust/issues/2645) and [figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=2700-17176&mode=design&t=UaNFF8R1A4ti0dQ0-4)

NOTE: shipping the PR would have **no user facing impact** since the Assistant Details modal is not yet accessible anywhere in the app. This is a prep for AssistantGallery Modal & My assistants pages that I'll do next.

## Priorization & product questions
- Assistant instructions
  - Since prompt can be very long (e.g. help see screenshot below) and summarizing is non trivial (store? or latency to display window?) and non essential, settled for scroll `auto` with a `max-h-64` as v1
  - Do we show prompts of system agents here? (@help in particular) => would say no, and just answer "Instructions of this assistant are not available." WDYT?
- assistant deletion -> ux wise it is shown for any assistant in the user's list -- but not if it's not in their list, to minimize mishaps
  - Would we prefer to always show it but disabled when deletion is not possible?
  - There can be user frustrations when an assistant they created and published is deleted by somebody else => just to confirm we are clear on that

- global assistants => at the time they are considered on the user's list, can only be triggered on / off for the whole workspace
  - allowing users to opt out of them (e.g. opt out of @gpt4) is slightly more work + I'm not sure we want that?

- Add / remove from my list
  - Only show for 'workspace' / 'published'
  - Not operational as per this PR

#### Not yet done - to priorize
- modal display: removing the blur
- modal closing: replacing the cross on right by a chevron on left
- display usage stats in the modal
- duplicate button (planned soon though)

To discuss whether:  should be done before user-facingness / ok v2 but guaranteed in climb / ok v3, should be done but we can wait  / actually not that important

#### Shots
With dust app action, already in list, deletable
image

Not yet in list, deletable, no action
image

Data sources
image

Delete popup
image

Long instructions
image

Raw model
image

### CODE REVIEW

File: front/components/assistant/AssistantDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+  return (
+    <div className="flex flex-col gap-1">
+      {dataSourceConfigurations.sort(compareDatasourceNames).map((ds) => {
+        const providerName = getProviderName(ds);
+        const DsLogo = providerName
+          ? CONNECTOR_CONFIGURATIONS[providerName].logoComponent
+          : CloudArrowDownIcon;
+        const dsDocumentNumberText = `(${
+          ds.filter.parents?.in.length ?? "all"
```
Comment (spolu) [inline front/components/assistant/AssistantDetails.tsx:139]: there's a word missing here no in case there are parents selection?

**Review by spolu — DISMISSED** [review]
LGTM

**Comment by spolu** [comment]
Since this is not user visible. The answer to all your questions are: this is fine, we can finetune as we go :)

**Comment by philipperolet** [comment]
> Since this is not user visible. The answer to all your questions are: this is fine, we can finetune as we go :)

Excellent :) I'll merge -- but the user visibleness will hopefully come very soon!

File: front/components/assistant/AssistantDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        } elements)`;
```
Comment (philipperolet) [inline front/components/assistant/AssistantDetails.tsx:140]:
```suggestion
        } element(s))`;
```

**Comment by Duncid** [comment]
How can I test these?
Visually, we're not done yet, needs a bit of love.

**Comment by philipperolet** [comment]
Yep, true -- cf the list of notes in the PR summary above, glad to have your opinion on the best tradeoffs

You can test on front-edge in ~5mns (deploying) => go to the existing assistants page, and click on the assistants descriptions to make the modal appear

**Pull request merged**